recipes/default.rb

```diff
@@ -10,26 +10,14 @@
   compile_time false if respond_to?(:compile_time)
 end
 
-if platform_family? 'rhel'
-  yum_repository 'influxdb' do
-    description 'InfluxDB Repository - RHEL \$releasever'
-    baseurl node['influxdb']['upstream_repository']
-    gpgkey 'https://repos.influxdata.com/influxdb.key'
+influxdb_install 'influxdb' do
+  if node['kernel']['machine'].include? '64'
+    arch_type 'amd64'
+  elsif node['kernel']['machine'].include? '386'
+    arch_type 'i386'
+  else
+    Chef::Log.warning "I am unable to determine arch type for: #{node['kernel']['machine']}"
   end
```

Comment:
Maybe I'm missing something, but it seems like a problem with chef-client if the detection is wrong and I would think lots of other things would break in that case as well, just like we trust in `node.platform_family`.  My preference is to have it in the provider, but I'll let @bdangit weigh in on this one.  

I also wonder if there's a more robust match than just an include as well?  It would be particularly nice if we could just use `node['kernel']['machine']` but it looks like that won't always match in the apt repo. 🎱